### ISSUE TITLE
Preconditions.checkNotNull on primitif type.

### ISSUE THREAD

**Issue by julesmougin** [immutables/immutables#285]
We are working with Immutable (org.immutables) and errorprone (com.google.errorprone).
                                                                                
During the upgrade of deps, we are stuck at Immutable:2.1.8.                    
                                                                                
Because in 2.1.9 (and greater) Optional attibute like Optional\<Integer> or Optional\<Boolean> (I don't check for other Optional\<String>, Optional\<Char>, etc...) generated classes introduce checkNotNull call on primitive type  which causes an errorprone error (PreconditionsCheckNotNullPrimitive).

example : 
```
@Value.Immutable
public interface Test {
  Optional<Boolean> isTrue();
}
```
generates this in 2.1.14
```
    /**
     * Initializes the optional value {@link Test#isTrue() isTrue} to isTrue.
     * @param isTrue The value for isTrue
     * @return {@code this} builder for chained invocation
     */
    public final Builder isTrue(boolean isTrue) {
      this.isTrue = Preconditions.checkNotNull(isTrue, "isTrue");
      return this;
    }

    /**
     * Initializes the optional value {@link Test#isTrue() isTrue} to isTrue.
     * @param isTrue The value for isTrue
     * @return {@code this} builder for use in a chained invocation
     */
    public final Builder isTrue(Optional<Boolean> isTrue) {
      this.isTrue = isTrue.orElse(null);
      return this;
    }
```

I have tried to pass @SuppressWarnings("PreconditionsCheckNotNullPrimitive") (cf. erroprone suggestion: http://errorprone.info/bugpattern/PreconditionsCheckNotNullPrimitive) with @Value.Style(passAnnotations) or use -XepDisableWarningsInGeneratedCode flag (cf. issue suggestion : https://github.com/google/error-prone/issues/329) but both solutions don't work.

I don't succes to find my @SuppressWarnings on gereted class...

**Comment by tbroyer** [comment]
Hmm, `-XepDisableWarningsInGeneratedCode` doesn't work because the `PreconditionsCheckNotNullPrimitive` check is an error by default; but it should work if tuning it down to a warning: `-Xep:PreconditionsCheckNotNullPrimitive:WARN`.
That'd still only be a workaround though; clearly, generating such code is a bug in Immutables to begin with.

**Comment by ldriscoll** [comment]
@julesmougin can you try adding this to your `@Value.Style()` - `optionalAcceptNullable = true` This makes it unwrap the boolean setter correctly.

**Comment by tbroyer** [comment]
That fixes it, thanks (I work with @julesmougin)

**Comment by julesmougin** [comment]
Ty for your help. It's working as @tbroyer said.

**Comment by elucash** [comment]
```java
  public final Builder isTrue(boolean isTrue) {
      this.isTrue = Preconditions.checkNotNull(isTrue, "isTrue");
      return this;
  }
```
Should be fixed anyway. Thank you for spotting this!

**Comment by tbroyer** [comment]
@elucash: I think we've had this discussion before, but using [error-prone](http://errorprone.info) when building/testing Immutables would probably not be a bad idea ;-)
(also, update to maven-compiler-plugin 3.5.1 which fixes a lot of bugs related to annotation processing; last "usable" version was 3.1 as far as I'm concerned)

**Comment by elucash** [comment]
@tbroyer we use error prone for the "fixture" compilation, even upgraded recently in [0b59a5b7](https://github.com/immutables/immutables/commit/0b59a5b73daaf8a2fc5b03119c835118f1ae1f88)

**Comment by elucash** [comment]
`maven-compiler-plugin 3.5.1` interesting, have to try it )

**Comment by tbroyer** [comment]
> we use error prone for the "fixture" compilation

Oh great! I had only looked at the root POM.

**Comment by tbroyer** [comment]
BTW, looking quickly at value-fixture, it seems like this bug might be triggered by the use of Java 8's `Optional` without `jdkOnly` (I see `GetterEncloser` is using an `Optional<Integer>` from Guava, and `UsingAllOptionals` is using an `Optional<Integer>` from the JDK but with `jdkOnly=true`)

**Comment by tbroyer** [comment]
Oh, actually, the `jdkOnly=true` generates a useless `Objects.requireNonNull` check too, it just isn't picked up by Error Prone (yet!):

```java
    /**
     * Initializes the optional value {@link UsingAllOptionals#v2() v2} to v2.
     * @param v2 The value for v2
     * @return {@code this} builder for chained invocation
     */
    public final Builder v2(int v2) {
      this.v2 = Objects.requireNonNull(v2, "v2");
      return this;
    }
```

**Comment by elucash** [comment]
wow, we got bug parade!

**Comment by elucash** [comment]
Just released 2.1.15 with the fix. Please, evaluate it. Thank you for reporting this, I also found a lot of small sibling bugs/inconsistencies around this piece of functionality and hope now it's much better!

**Comment by julesmougin** [comment]
I build and run with 2.1.15 it's look like to work :+1:
